hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestSourceFormatAdapter.java

```diff
@@ -136,16 +146,23 @@ public void testJsonSanitization(String unsanitizedDataFile, String sanitizedDat
 
   public static class TestRowDataSource extends Source<Dataset<Row>> {
     private final InputBatch<Dataset<Row>> batch;
+    private final SanitizationUtils rowSanitizer;
 
     public TestRowDataSource(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
                              SchemaProvider schemaProvider, InputBatch<Dataset<Row>> batch) {
       super(props, sparkContext, sparkSession, schemaProvider, SourceType.ROW);
       this.batch = batch;
+      this.rowSanitizer = new SanitizationUtils();
     }
 
     @Override
     protected InputBatch<Dataset<Row>> fetchNewData(Option<String> lastCkptStr, long sourceLimit) {
-      return batch;
+      return batch.getBatch().map(dsr -> {
+        Dataset<Row> tranformed = SanitizationUtils.sanitizeColumnNamesForAvro(dsr, getInvalidCharMask(props));
```

Comment:
Why do we want the sanitizer here? Shouldn't we be checking that the source format adapter is doing the sanitization?

Reply:
or should we have this test class just override `fetchNextBatch`?

Reply:
We are bypassing RowSource in Unit test . 
SourceFormatAdapter is not doing any sanitization .

Reply:
added to cover actual RowSource .